tests/test_minimizer.py

```diff
@@ -41,3 +44,26 @@ def test_equilibrate_host():
     assert coords.shape[0] == host_coords.shape[0] + mol.GetNumAtoms()
     assert coords.shape[1] == host_coords.shape[1]
     assert box.shape == host_box.shape
+
+
+def test_local_minimize_water_box():
+    """
+    Test that we can locally relax a box of water by selecting some random indices.
+    """
+
+    system, x0, box0, _ = builders.build_water_system(4.0)
+    x0 = to_md_units(x0)
+    lamb = 0.0
+    bps, _ = openmm_deserializer.deserialize_system(system, cutoff=1.2)
+    box0 += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes at the boundary
+
+    val_and_grad_fn = minimizer.get_val_and_grad_fn(bps, box0, lamb)
+
+    free_idxs = [0, 2, 3, 6, 7, 9, 15, 16]
+    frozen_idxs = set(range(len(x0))).difference(set(free_idxs))
+    frozen_idxs = list(frozen_idxs)
+
+    x_opt = minimizer.local_minimize(x0, val_and_grad_fn, free_idxs)
+
+    np.testing.assert_array_equal(x0[frozen_idxs], x_opt[frozen_idxs])
+    assert np.linalg.norm(x0[free_idxs] - x_opt[free_idxs]) > 0.01
```

Comment:
nit: worth checking that the energy has decreased?

Reply:
oops, never mind, I see that we already have an assertion for this in the minimizer.